packages/repository-json-schema/src/build-schema.ts

```diff
@@ -142,16 +151,24 @@ export function metaToJsonProperty(meta: PropertyDefinition): JSONSchema {
  * reflection API
  * @param ctor Constructor of class to convert from
  */
-export function modelToJsonSchema(ctor: Function): JSONSchema {
+export function modelToJsonSchema(
+  ctor: Function,
+  options: JsonSchemaOptions = {},
+): JSONSchema {
+  options.visited = options.visited || new Set<string>();
```

Comment:
Please don't modify original `options`.

```suggestion
  if (!options.visited) options = {...options, visited: new Set<string>()};
```

Please add a test to verify.

Reply:
> Please don't modify original options.

Hmm, any reason?

Reply:
It's confusing for callers when you modify the `options` object. The caller may use the same `options` instance for multiple calls.

For example:

```ts
const SCHEMA_OPTIONS = {includeRelations: true};

const productSchema = getJsonSchema(Product, SCHEMA_OPTIONS);
const categorySchema = getJsonSchema(Category, SCHEMA_OPTIONS);
```

Reply:
👍 got it thank you.